tsdb/head_test.go

```diff
@@ -1810,3 +1810,105 @@ func newTestHead(t testing.TB, chunkRange int64, compressWAL bool) (*Head, *wal.
 		testutil.Ok(t, os.RemoveAll(dir))
 	}
 }
+
+func TestHeadLabelNamesValuesWithMinMaxRange(t *testing.T) {
+	head, _, closer := newTestHead(t, 1000, false)
+	defer closer()
+	defer func() {
+		testutil.Ok(t, head.Close())
+	}()
+
+	const (
+		firstSeriesTimestamp  int64 = 100
+		secondSeriesTimestamp int64 = 200
+		lastSeriesTimestamp   int64 = 300
+	)
+	var (
+		seriesTimestamps = []int64{firstSeriesTimestamp,
+			secondSeriesTimestamp,
+			lastSeriesTimestamp,
+		}
+		expectedLabelNames  = []string{"a", "b", "c"}
+		expectedLabelValues = []string{"d", "e", "f"}
+	)
+
+	app := head.Appender()
+	for i, name := range expectedLabelNames {
+		_, err := app.Add(labels.Labels{{Name: name, Value: expectedLabelValues[i]}}, seriesTimestamps[i], 0)
+		testutil.Ok(t, err)
+	}
+	err := app.Commit()
+	testutil.Ok(t, err)
+
+	testutil.Assert(t,
+		head.MinTime() == firstSeriesTimestamp,
+		"the min timestamp of the head should be equal to the timestamp of first series added",
+	)
+	testutil.Assert(t,
+		head.MaxTime() == lastSeriesTimestamp,
+		"the max timestamp of the head should be equal to the timestamp of last series added",
+	)
+
+	// maxt is less than head min and mint is less than head max
```

Comment:
I would be nitty picky here.. I think this deserves table test (: Essentially there are many repetitions here and table test would fit here perfectly. What do you think? (: 

I always have this template for that:
```
for _, tcase := range []struct{
  name string 
  // other things
}{
    { 
      // cases
    },
}{
  t.Run(tcase.name, func(t *testing.T) {
            // test
  })
}
```

Reply:
+1, and the test cases for empty returns are a bit weird. We could assume mint <= maxt and just test these 4 cases

```
1
      |------|
|---|

2
      |------|
    |---|

3
      |------|
           |---|

4
      |------|
               |---|
```

Reply:
@codesome I picked those test cases since they execute all 4 cases of the `if` block i added here https://github.com/prometheus/prometheus/pull/7288/files#diff-6fb13507bfadf9819fea5bda61f599e6R1565